libs/antsi/src/color.rs

```diff
@@ -0,0 +1,313 @@
+/// `COLORTERM` environment variable to inspect if truecolor is supported. (the value of the
+/// variable will be `truecolor`). Most built-terminals, do **not** support true-color by default.
+/// To programmatically check for support it is advised to use a crate similar to [`supports-color`](https://lib.rs/crates/supports-color)
+///
+/// ## Specification
+///
+/// This mode was initially specified in 1994, together with [`IndexedColor`], but most terminals do
+/// not support the format outlined in [ISO 8613-6], The standard is a lot more complete than the
+/// now used format of `ESC[38;2;{r};{g};{b}m` and featured optional color space id, tolerance and
+/// color space associated with the tolerance parameters.
+///
+/// Only `xterm` supports this scheme, where the color space id and tolerance parameters are
+/// ignored. The specification also uses `:` as a separator instead of `;`. `xterm` was the first
+/// terminal with wide-spread adoption that implemented RGB support in 2012, and all other terminals
+/// copied their implementation. For the reason why there's a discrepancy between the standard and
+/// implementation a detailed reasoning is provided in the [xterm repo](https://github.com/ThomasDickey/xterm-snapshots/blob/8d625aa49d5fdaa055a9f26d514121f032c7b771/charproc.c#L1957-L2028)
+///
+/// [ISO 8613-6] also specified multiple additional modes, like transparent, CMK, and CMYK support.
+///
+/// [ISO 8613-6]: https://www.iso.org/standard/22943.html
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct RgbColor {
+    r: u8,
+    g: u8,
+    b: u8,
+}
+
+impl RgbColor {
+    /// Create a new RGB color
+    #[must_use]
+    pub const fn new(r: u8, g: u8, b: u8) -> Self {
+        Self { r, g, b }
+    }
+}
+
+/// Collection of every possible terminal color supported by `antsi`
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Color {
+    /// Basic 8 color palette
+    ///
+    /// Supported by pretty much every terminal, for more information see [`BasicColor`]
+    Basic(BasicColor),
+
+    /// Bright variant of the basic color palette
+    ///
+    /// Supported by pretty much every terminal, for more information see [`BrightColor`]
+    Bright(BrightColor),
+
+    /// Indexed color, more commonly known as xterm-color
+    ///
+    /// Supported by pretty much every terminal since the early 2000s, for more information see
+    /// [`IndexedColor`]
+    Indexed(IndexedColor),
+
+    /// 24bit color support, more commonly known as truecolor
+    ///
+    /// Supported by some modern terminals since 2015+, for more information see [`RgbColor`]
+    Rgb(RgbColor),
+}
+
+impl Color {
+    /// Return [`BasicColor`], if value of variant [`Color::Basic`]
+    #[must_use]
+    pub const fn as_basic(&self) -> Option<BasicColor> {
+        if let Self::Basic(color) = self {
+            Some(*color)
+        } else {
+            None
+        }
+    }
```

Comment:
As `match` statements are allowed in `const` context, do we really want these methods?

Reply:
sure, tho I added them as convenience getters (to e.g. allow easy chaining), if you think that is not valuable I'll remove them :D

Reply:
I think the signatures are misleading. From reading they suggest, that they may get converted into `BasicColor` and return `None` if there is no representation for that.